cookbook/asset/custom_version_strategy.rst

```diff
@@ -0,0 +1,264 @@
+.. index::
+   single: Asset; Custom Version Strategy
+
+How to use a Custom Version Strategy for Assets
+===============================================
+
+Symfony by default does not perform asset versioning. You can specify the
+:ref:`version <reference-framework-assets-version>` and
+:ref:`version_format <reference-framework-assets-version-format>` configuration
+options to add a simple version to all assets (or a specific set of assets
+grouped as a :ref:`package <reference-framework-assets-packages>`).
+
+.. versionadded:: 2.7
+    The Asset component was introduced in Symfony 2.7.
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # app/config/config.yml
+        framework:
+            assets:
+                version: "20150530"
+                version_format: "%%s?version=%%s"
+
+    .. code-block:: xml
+
+        <!-- app/config/config.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:framework="http://symfony.com/schema/dic/symfony"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/symfony
+                http://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
+
+            <framework:config>
+                <framework:assets
+                    version="20150530"
+                    version-format="%%s?version=%%s" />
+            </framework:config>
+        </container>
+
+    .. code-block:: php
+
+        // app/config/config.php
+        $container->loadFromExtension('framework', array(
+            'assets' => array(
+                'version' => '20150530',
+                'version_format' => '%%s?version=%%s',
+            ),
+        ));
+
+However, if you require more control, you need to create a custom version
+strategy.
+
+Default Package
+---------------
+
+The default package is used when you do not specify a package name in the
+:ref:`asset <reference-twig-function-asset>` Twig function. In order to
+override the version strategy used by the default package, it is necessary
+to add a compiler pass.
+
+This example shows how to integrate with `gulp-buster`_.
+
+.. note::
+
+    busters.json as referenced below is the output from gulp-buster which
+    maps each asset file to its hash. A small snippet of the file's format
+    (JSON object):
+
+.. code-block:: json
+
+    {
+        "js/script.js": "f9c7afd05729f10f55b689f36bb20172",
+        "css/style.css": "91cd067f79a5839536b46c494c4272d8"
+    }
```

Comment:
You need to indent the whole code block by four spaces so that it will be rendered inside the note.